File: Manvel/Cs50/Week_1_continue/vigenere.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>

int main (void)
{
string s = get_string ("input sentance: ");
string key = get_string ("input key: ");
int i;
int j=0;
int o;
int p;
int n = strlen(s);
int m = strlen(key);

for(i=0;i<n;i++)
{
     p = (int)key[j];

    if((int)s[i]>=65 && (int)s[i]<=90)
    {
        o=((int)s[i] + p) -97 ;
        j++; 

        if(o>90)
        {
            o-=26;
        }  
    }
    else
    {
        if((int)s[i]>=97 && (int)s[i]<=122)
        {
            o=(int)s[i] + p -97 ;
            j++;
            if(o>122)
            {
                o-=26;
            }
         }
        else
        {
            o=(int)s[i];
        }
    }

    if(j==m)

    {
        j=0;
    }
    
    s[i] = (char)o;
}
        printf("%s\n",s);

}

File: HaykAmirjanyan/pset6/vigenere/vigenere.py
import sys
from sys import argv
from cs50 import get_string

if len(sys.argv) > 1:
    key = argv[1]
else:
    print("enter the key:")
    exit(1)

for i in range(len(key)):
    if not key[i].isalpha():
        print("Enter a valid key")
        exit(1)

key = key.upper()
sentance = get_string("Wriite a sentance: ")
j = 0
out_sentance = ""

for l in range(len(sentance)):
    m = ord(sentance[l])
    if sentance[l].isalpha():
        m = ord(sentance[l]) + ord(key[j]) - 65
        j += 1
    if j == len(key):
        j = 0
    if (sentance[l].islower() and m > 122) or (sentance[l].isupper() and m > 90):
        m -= 26
    out_sentance += chr(m)
print("ciphertext:" + out_sentance)